NEWS.md

```diff
@@ -465,6 +465,8 @@ Deprecated or removed
     `prompt_if_incorrect` argument are deprecated. Instead, prompting behavior is controlled using
     the `allow_prompt` keyword in the `LibGit2.CredentialPayload` constructor ([#23690]).
 
+  * `gradient` is deprecated and will be removed in the next release ([#23816]).
```

Comment:
I think "will be removed in the next release" is redundant with "deprecated," since we always remove deprecated things in the subsequent release. Probably doesn't matter too much though.